### PR TITLE
[Feature] Created new schema with Events

### PR DESCRIPTION
**Pull request by nup9151f**
## Summary

This PR updates the schema to use Event entity and new Transaction types based on function called.

## Description

Changes 

- New Event entity has been added with a many-to-one relationship with Transaction.
- New Stats for Events have been added, these stats do not have a `gasUsed` attribute because this is an attribute that belongs to the transaction.
- Current Transaction types are now Event types.
- New Transaction types have been added based on the function called in the transaction. We get the function name using a lookup table and the function selector from the transaction input.
- Transaction stats are collected for all known transaction types. In case of an unknown transaction type we save the transaction but do not collect Transaction Stats for it.
- Event stats are always collected given that we do not use these for charges.

## Type of Change

Please check the boxes that apply to your pull request:

- [X] Breaking change (fix or feature that would cause existing functionality to change)

Breaking changes:
- Transaction field `type` renamed to `transactionType`
- Values for transaction types have changed.

### CODE REVIEW

**Review by tinypell3ts — APPROVED** [review]

File: schema.graphql [hunk 1]
```diff
@@ -112,4 +114,106 @@ type TransactionUserAppStat @entity {
   createdAtBlock: BigInt!
   updatedAt: BigInt!
   updatedAtBlock: BigInt!
+}
+
+type Event @entity(immutable: true) {
+  id: ID!
+  transaction: Transaction
+  timestamp: Timestamp!
```
Comment (george-openformat) [inline schema.graphql]: `timestamp` and `createdAt` are the same. Should timestamp be removed to keep wording consistent?

File: schema.graphql [hunk 2]
```diff
@@ -7,10 +7,12 @@ type User @entity(immutable: true) {
 type Transaction @entity(immutable: true) {
   id: ID!
   timestamp: Timestamp!
```
Comment (george-openformat) [inline schema.graphql]: Just noticed timestamp here as well. Should we remove and just use createdAt as they are the same?

**Review by george-openformat — APPROVED** [review]
Loving it! This feels much better

File: src/helpers/transactions.ts [hunk 3]
```diff
@@ -1,46 +1,62 @@
...
+export const BADGE_CREATE_TYPE                = "Badge Create";
+export const BADGE_MINT_TYPE                  = "Badge Reward";
+export const BADGE_TRANSFER_TYPE              = "Badge Transfer";
+
+export const APP_CREATE_TYPE                  = "App Create";
+
+// Transaction Types
+// Keys are function selectors from transaction.input
+const functionTypes = new TypedMap<string, string>();
+functionTypes.set("0xb92bafd6", "Batch Mint Badge");
```
Comment (george-openformat) [inline src/helpers/transactions.ts:39]: After playing with this I realise I left out `create` function in the app factory contract.
Could you please add:

`functionTypes.set("0x663ff5a9", "Create App");`

File: schema.graphql (hunk 1 above)
Comment (nup9151f) [reply schema.graphql]: Agree! I am deleting `timestamp` and leaving `createdAt`.

File: schema.graphql (hunk 2 above)
Comment (nup9151f) [reply schema.graphql]: Yes! Deleted as in Events.

File: src/helpers/transactions.ts (hunk 3 above)
Comment (nup9151f) [reply src/helpers/transactions.ts:39]: Added! :+1:

**Pull request merged**